kbfsgit/runner.go

```diff
@@ -1501,17 +1503,49 @@ func (r *runner) pushSome(
 			err = nil
 		}
 
+		// TODO (KBFS-2682): also load parent refs.
+		refs := make([]*plumbing.Reference, 0, len(refspecs))
+
 		// All non-delete refspecs in the batch get the same error.
 		for _, refspec := range refspecs {
 			refStr := refspec.String()
 			start := strings.Index(refStr, ":") + 1
 			dst := refStr[start:]
 			results[dst] = err
+			ref, refErr := repo.Storer.Reference(plumbing.ReferenceName(dst))
```

Comment:
`repo.Storer` will give you the commits that are already in KBFS, not the ones being pushed from the local repo.  I think you'll need to make a new `Repository` object using a `billy.osfs` storage layer pointing to the local repo, to find the references being pushed.

Reply:
At this point they should already be in KBFS, since we've already done the `FetchContext`. Am I missing something?

Reply:
I'll take your suggestion anyway and move the delta derivation to before the `FetchContext`, since otherwise calculating the delta is harder. But I still want to make sure that my understanding is correct.

Reply:
Ah I see, yeah I missed that it was after the FetchContext.  You're right.  But yeah, you should do it before, so you can figure out what changed.

Reply:
Thanks! I hope to have a final review ready for you tomorrow.